### PR TITLE
Simplify timer with macrotask callback

### PR DESCRIPTION
**Pull request by kevinkassimo**
Instead of using microtask arrangement magic, explicitly notify processing of macrotasks (timers) with callback.

This makes the implementation of timer a bit simpler. It also corrects a less obvious bug of scheduling:
```js
setTimeout(() => {
  console.log("s1");
  Promise.resolve().then(() => {
    console.log("r11");
  }).then(() => {
    console.log("r12");
  });
}, 0);
setTimeout(() => {
  console.log("s2");
  Promise.resolve().then(() => {
    console.log("r21");
  }).then(() => {
    console.log("r22");
  });
}, 0);
```
Before (problem: double `.then`s cannot be correctly ordered):
```
s1
r11
s2
r12
r21
r22
```
After (same behavior as in browser):
```
s1
r11
r12
s2
r21
r22
```

### CODE REVIEW

File: cli/js/web/timers.ts [hunk 1]
```diff
@@ -31,8 +31,15 @@ function clearGlobalTimeout(): void {
 
 let pendingEvents = 0;
 const pendingFireTimers: Timer[] = [];
-let hasPendingFireTimers = false;
-let pendingScheduleTimers: Timer[] = [];
+
+// Handler when timer macrotask is should be called
```
Comment (kitsonk) [inline cli/js/web/timers.ts]: JSDoc?  Even though this is internal, JSDoc does flow through better in editors that support it.

**Comment by ry** [comment]
@kevinkassimo Sorry for the delay in review. I'm a little hesitant to add non-op bindings, but if it's fixing a bug and simplifying things that's good. Can you add the test you mentioned?

**Comment by bartlomieju** [comment]
While we are changing how timers are implemented we could think about the solution of async op leaking in case of timers.

https://github.com/denoland/deno/blob/54d1f299dc774c695056f04cfe1013287b53b86a/cli/js/tests/signal_test.ts#L106-L138

(comment at the end of snippet)

**Comment by kevinkassimo** [comment]
@bartlomieju I think that is technically not a bug a timer itself, and I also found it not quite straightforward. One way to work around this is to always add a `await new Promise(res => setTimeout(res, 0))` to force a poll before asserting resources.

**Comment by bartlomieju** [comment]
> @bartlomieju I think that is technically not a bug a timer itself, and I also found it not quite straightforward. One way to work around this is to always add a `await new Promise(res => setTimeout(res, 0))` to force a poll before asserting resources in `unitTest`.

Now that you said it seems pretty obvious... I'll try to rewrite it later then.

**Pull request merged**